spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/ChannelNode.java

```diff
@@ -21,21 +21,18 @@
 /**
  * @author Andy Clement
  * @author David Turanski
+ * @author Ilayaperumal Gopinathan
  */
 public class ChannelNode extends AstNode {
 
-	private ChannelType channelType;
+	private String nameComponent;
 
-	private List<String> nameComponents;
+	private List<String> labelComponents;
```

Comment:
Given we aren't using the indexes inside the stream definition, they are just the components after DOT token and there is no impact even if the named channel has `.` in it. Perhaps, we need a better name here?